libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/LocationComponent.kt

```diff
@@ -0,0 +1,83 @@
+package com.mapbox.navigation.ui.maps.internal.ui
+
+import android.animation.ValueAnimator
+import com.mapbox.geojson.Point
+import com.mapbox.maps.MapView
+import com.mapbox.maps.plugin.LocationPuck
+import com.mapbox.maps.plugin.locationcomponent.LocationConsumer
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.extensions.flowLocationMatcherResult
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+@ExperimentalPreviewMapboxNavigationAPI
+class LocationComponent(
+    private val mapView: MapView,
+    private val locationPuck: LocationPuck,
+    private val locationProvider: NavigationLocationProvider,
+    private val enableLocationUpdates: Boolean = true
```

Comment:
These option parameters are not good for java backwards compatibility.

What do you think about moving the option as a function?

```
class LocationComponent(
...

  fun enableLocationUpdates(enable: Boolean)
```

Reply:
Otherwise it would need to go in an `Options.Builder`

Reply:
I wouldn't worry about the "java backwards compatibility" (this is an experimental, internal component). 
Sure, we can do the method instead.

Reply:
> I wouldn't worry about the "java backwards compatibility

We're addressing java backwards compatiblity https://github.com/mapbox/mapbox-navigation-android/pull/6292. That is part of creating a stable api. If there is no intention to make this LocationComponent public, then sure we don't have to worry about it.

[Some version of this](https://github.com/mapbox/mapbox-navigation-android/blob/456bcf818641bc1a42edc9e27436a1e54ac6c4db/libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/location/CarLocationRenderer.kt#L15-L29) is going to become a public api for AA

Reply:
Nvm. I found a better way to deal with this.
I've split `LocationComponent` into two separate components:
- **LocationComponent** responsible for registering `LocationObserver` and updating `NavigationLocationProvider`
- **LocationPuckComponent** for configuring and updating LocationPuck. This component depends on `MapboxMap`,`LocationComponentPlugin`, `LocationPuck` and `NavigationLocationProvider`, which are available in both AA and DropIn UI.